packages/container/lib/registry.js

```diff
@@ -691,6 +692,32 @@ Registry.prototype = {
     });
   },
 
+  /**
+   @method eachForType
+   @param {String} type the type to iterate over
+   @param {Function} callback the function to invoke for each item
+   @private
+  */
+  eachForType(type, callback) {
+    let registeredNames = keys(this.registrations);
+    for (let index = 0, length = registeredNames.length; index < length; index++) {
+      let fullName = registeredNames[index];
+      let itemType = fullName.split(':')[0];
+
+      if (itemType === type) {
+        callback(fullName, this.registrations[fullName]);
+      }
+    }
+
+    if (this.fallback) {
+      this.fallback.eachForType(type, callback);
+    }
+
+    if (this.resolver.eachForType) {
+      this.resolver.eachForType(type, callback);
```

Comment:
Yes, it is theoretically possible for things to show up twice, but unlikely. 

* We only walk manually registered items in the registry (why manually register if it will auto-resolve?). 
* Book keeping to prevent double entry seems unneeded as we are keeping it in a small dictionary which will dedupe for us.